### PR TITLE
Add map of people to publiclab.org/people #1940

### PR DESCRIPTION
**Pull request by sagarpreet-chadha**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

**Solves** #1940 . 
@jywarren , @ebarry , kindly review this PR . Thanks !

screen shot 2018-01-29 at 12 02 48 am

* [X] all tests pass -- `rake test:all`
* [X] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [X] pull request is descriptively named with #number reference back to original issue

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

File: app/services/search_service.rb [hunk 1]
```diff
@@ -219,4 +219,30 @@ def nearbyNodes(srchString)
     sresult
   end
 
+#GET X number of latest people/contributors 
+# X = srchString
+def recentPeople(srchString)
+      sresult = DocList.new
+      drupalusers = DrupalUser.select('*, MAX(node.changed) AS last_updated')
```
Comment (jywarren) [inline app/services/search_service.rb]: is it possible to use User here and not DrupalUser? Sorry, just trying to reduce drupaluser usage as much as possible! We can `includes` drupaluser here if need be, but thinking of the minimal use of DrupalUser until we finally merge that into User.

**Comment by jywarren** [comment]
This looks awesome. I'd like to ask if we could add a unit test for the Service changes, and a functional test for the API query. Thanks so much @sagarpreet-chadha -- this is amazing!

File: app/services/search_service.rb (hunk 1 above)
Comment (sagarpreet-chadha) [reply app/services/search_service.rb]: Okay...makes sense 👍 👍 .

File: app/views/map/_peopleLeaflet.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  #map<%= unique_id %> { width:100%; height:300px; margin: 0; position: relative;}
+</style>
+<div class="leaflet-map" id="map<%= unique_id %>"></div>
+<% if defined? people %><p><i><small>Share your own location on <a href='/profile'>your profile</a>.</small></i></p><% end %>
+<script>
+  var map<%= unique_id %> = L.map('map<%= unique_id %>').on('load', onMapLoad).setView([<%= lat %>,<%= lon %>], 2);
+  L.tileLayer("//a.tiles.mapbox.com/v3/jywarren.map-lmrwb2em/{z}/{x}/{y}.png").addTo(map<%= unique_id %>);
+  function onMapLoad(e){
+ 	$.getJSON("/api/srch/peoplelocations?srchString=100" , function(data){
+	 			if (!!data.items){
```
[the remark this answers is not in the document]
Comment (sagarpreet-chadha) [reply app/views/map/_peopleLeaflet.html.erb]: Okay :-) Thank You !

**Comment by jywarren** [comment]
How does this look now? Good to go? Thanks!!!

**Comment by sagarpreet-chadha** [comment]
Yes i have done the changes and added the tests . Are the changes ok ? Thanks .

**Comment by jywarren** [comment]
This looks fantastic. Thanks so much. 

I'm going to merge it, but if you could try to maintain really consistent two-space tabbing in the future, that would be really great! I think the Rubocop style guide bot will start to catch this pretty soon, but until then we should try our best to keep our code formatting really clean and consistent. 

Thanks so much, @sagarpreet-chadha !! This is such an exciting feature :-) 🎉 🎈 💯

**Comment by sagarpreet-chadha** [comment]
Thank yo so much @jywarren !! Will take care of spacing from now on :-)

**Comment by jywarren** [comment]
https://publiclab.org/people -- is live!!!

This is great!

**Pull request merged**